## Introduction

In [4]:
%%sql

SHOW TABLES;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
12 rows affected.


[('auditor_report',),
 ('combined_analysis_table',),
 ('data_dictionary',),
 ('employee',),
 ('global_water_access',),
 ('incorrect_records',),
 ('location',),
 ('project_progress',),
 ('visits',),
 ('water_quality',),
 ('water_source',),
 ('well_pollution',)]

In [26]:
%%sql
SELECT *
FROM
    data_dictionary;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
49 rows affected.


[('employee', 'assigned_employee_id', 'Unique ID assigned to each employee', 'INT', 'visits'),
 ('employee', 'employee_name', 'Name of the employee', 'VARCHAR(255)', ''),
 ('employee', 'phone_number', 'Contact number of the employee', 'VARCHAR(15)', ''),
 ('employee', 'email', 'Email address of the employee', 'VARCHAR(255)', ''),
 ('employee', 'address', 'Residential address of the employee', 'VARCHAR(255)', ''),
 ('employee', 'town_name', 'Name of the town where the employee resides', 'VARCHAR(255)', ''),
 ('employee', 'province_name', 'Name of the province where the employee resides', 'VARCHAR(255)', ''),
 ('employee', 'position', 'Position or job title of the employee', 'VARCHAR(255)', ''),
 ('visits', 'record_id', 'Unique ID assigned to each visit', 'int', 'water_quality, water_source'),
 ('visits', 'location_id', 'ID of the location visited', 'varchar(255)', 'location'),
 ('visits', 'source_id', 'ID of the water source visited', 'varchar(510)', 'well_pollution'),
 ('visits', 'time_of_record', 'Date and time of the visit', 'datetime', ''),
 ('visits', 'visit_count', 'Number of visits made to this location', 'int', ''),
 ('visits', 'time_in_queue', 'Time spent by people waiting for water in a queue at the location', 'int', ''),
 ('visits', 'assigned_employee_id', 'ID of the employee who visited the location', 'int', 'employee'),
 ('water_quality', 'record_id', 'Unique ID assigned to each record', 'int', ''),
 ('water_quality', 'subjective_quality_score', 'Score representing the subjective quality of the water source', 'int', ''),
 ('water_quality', 'visit_count', 'Number of visits made for data collection', 'int', ''),
 ('water_source', 'source_id', 'Unique ID assigned to each water source', 'VARCHAR(510)', 'visits'),
 ('water_source', 'type_of_water_source', 'Type or category of the water source. Can be : tap_in_home, tap_in_home_broken, well, shared_tap, river', 'VARCHAR(255)', ''),
 ('water_source', 'number_of_people_served', 'Number of people served by this water source', 'INT', ''),
 ('well_pollution', 'source_id', 'ID of the water source being tested for pollution', 'varchar(258)', 'visits'),
 ('well_pollution', 'date', 'Date of the pollution test', 'datetime', ''),
 ('well_pollution', 'description', 'Description of the pollution test', 'varchar(255)', ''),
 ('well_pollution', 'pollutant_ppm', 'Result of the pollution test in parts per million', 'float', ''),
 ('well_pollution', 'biological', 'Biological contamination level', 'float', ''),
 ('well_pollution', 'results', "Result of the pollution test. Can be 'dirty', 'biologically contaminated', or 'clean'", 'varchar(255)', ''),
 ('location', 'location_id', 'Unique ID assigned to each location', 'varchar(255)', 'visits'),
 ('location', 'address', 'Address of the location', 'varchar(255)', ''),
 ('location', 'province_name', 'Name of the province where the location is situated', 'varchar(255)', ''),
 ('location', 'town_name', 'Name of the town where the location is situated', 'varchar(255)', ''),
 ('location', 'location_type', 'Type or category of the location', 'varchar(255)', ''),
 ('global_water_access', 'name', 'The country or area name', 'varchar(255)', ''),
 ('global_water_access', 'region', 'Geographical region', 'varchar(255)', ''),
 ('global_water_access', 'year', 'Year of the data record', 'int', ''),
 ('global_water_access', 'pop_n', 'The national population size estimate in thousands', 'float', ''),
 ('global_water_access', 'pop_u', 'The urban population share estimate in percentage points (%)', 'float', ''),
 ('global_water_access', 'wat_bas_n', 'The estimated national share of people with at least basic service (%)', 'float', ''),
 ('global_water_access', 'wat_lim_n', 'The estimated national share of people with limited service (%)', 'float', ''),
 ('global_water_access', 'wat_unimp_n', 'The estimated national share of people with unimproved service (%)', 'float', ''),
 ('global_water_access', 'wat_sur_n', 'The estimated national share of people with surfa

## 1. Cleaning the data

In [18]:
%%sql

SELECT * 
FROM
    employee 
LIMIT 5;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
5 rows affected.


[(0, 'Amara Jengo', '+99637993287', 'amara.jengo@ndogowater.gov', '36 Pwani Mchangani Road', 'Sokoto', 'Ilanga', 'Field Surveyor'),
 (1, 'Bello Azibo', '+99643864786', 'bello.azibo@ndogowater.gov', '129 Ziwa La Kioo Road', 'Kilimani', 'Rural', 'Field Surveyor'),
 (2, 'Bakari Iniko', '+99222599041', 'bakari.iniko@ndogowater.gov', '18 Mlima Tazama Avenue', 'Hawassa', 'Rural', 'Field Surveyor'),
 (3, 'Malachi Mavuso', '+99945849900', 'malachi.mavuso@ndogowater.gov', '100 Mogadishu Road', 'Akatsi', 'Lusaka', 'Field Surveyor'),
 (4, 'Cheche Buhle', '+99381679640', 'cheche.buhle@ndogowater.gov', '1 Savanna Street', 'Akatsi', 'Rural', 'Field Surveyor')]

In [ ]:
%%sql
# updated employees emails into required format

UPDATE
    employee
SET
    email = CONCAT((LOWER(REPLACE(employee_name, ' ', '.'))),'@ndogowater.gov');
    

    

 * mysql+pymysql://root:***@localhost:3306/md_water_services
56 rows affected.


In [ ]:
%%sql
# removed trailing spaces that add an extra character to phone number

UPDATE
    employee
SET
    phone_number = TRIM(Phone_number);



 * mysql+pymysql://root:***@localhost:3306/md_water_services
56 rows affected.


In [118]:
%%sql
# the employee emails are now updated in required format
# length of phone number is now 12, not 13.

SELECT
    Employee_name,
    Email,
    Phone_number,
    LENGTH(Phone_number) AS length_phone_no
FROM
    employee
LIMIT 5;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
5 rows affected.


[('Amara Jengo', 'amara.jengo@ndogowater.gov', '+99637993287', 12),
 ('Bello Azibo', 'bello.azibo@ndogowater.gov', '+99643864786', 12),
 ('Bakari Iniko', 'bakari.iniko@ndogowater.gov', '+99222599041', 12),
 ('Malachi Mavuso', 'malachi.mavuso@ndogowater.gov', '+99945849900', 12),
 ('Cheche Buhle', 'cheche.buhle@ndogowater.gov', '+99381679640', 12)]

## 2. Honouring the Workers

In [59]:
%%sql
#uncovered how many employees live in each town

SELECT
    Town_name,
    COUNT(Town_name) AS employee_town_count
FROM
    employee
GROUP BY Town_name;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
9 rows affected.


[('Ilanga', 3),
 ('Rural', 29),
 ('Lusaka', 4),
 ('Zanzibar', 4),
 ('Dahabu', 6),
 ('Kintampo', 1),
 ('Harare', 5),
 ('Yaounde', 1),
 ('Serowe', 3)]

In [371]:
%%sql
#identified the top 3 field surveyors who made the most visits by counting number of records each employee collected

SELECT
    assigned_employee_id,
    COUNT(record_id) AS total_visit_count
FROM
    visits
GROUP BY
    assigned_employee_id
ORDER BY
    COUNT(record_id) DESC
LIMIT 3;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
3 rows affected.


[(1, 3708), (30, 3676), (34, 3539)]

In [372]:
%%sql
#retrieved info of the top 3 employees for submission to Pres. Naledi

SELECT
    assigned_employee_id,
    employee_name,
    email,
    phone_number,
    position
FROM
    employee
WHERE
    assigned_employee_id IN ('1','30','34');

 * mysql+pymysql://root:***@localhost:3306/md_water_services
3 rows affected.


[(1, 'Bello Azibo', 'bello.azibo@ndogowater.gov', '+99643864786', 'Field Surveyor'),
 (30, 'Pili Zola', 'pili.zola@ndogowater.gov', '+99822478933', 'Field Surveyor'),
 (34, 'Rudo Imani', 'rudo.imani@ndogowater.gov', '+99046972648', 'Field Surveyor')]

## 3. Analysing Locations

In [126]:
%%sql
#counted number of records per town

SELECT
    town_name,
    COUNT(town_name) AS records_per_town
FROM
    location
GROUP BY
    town_name
ORDER BY 
    (COUNT(town_name)) DESC
LIMIT 5;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
5 rows affected.


[('Rural', 23740),
 ('Harare', 1650),
 ('Amina', 1090),
 ('Lusaka', 1070),
 ('Mrembo', 990)]

In [128]:
%%sql
#counted number of records per province

SELECT
    province_name,
    COUNT(province_name) AS records_per_province
FROM
    location
GROUP BY
    province_name
ORDER BY 
    (COUNT(province_name)) DESC;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
5 rows affected.


[('Kilimani', 9510),
 ('Akatsi', 8940),
 ('Sokoto', 8220),
 ('Amanzi', 6950),
 ('Hawassa', 6030)]

In [132]:
%%sql
#uncovered the record count of each town within each province

SELECT
    province_name,
    town_name,
    COUNT(town_name) AS records_per_town
FROM
    location
GROUP BY
    province_name,
    town_name
ORDER BY 
    province_name,
    (COUNT(province_name)) DESC
LIMIT 5;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
5 rows affected.


[('Akatsi', 'Rural', 6290),
 ('Akatsi', 'Lusaka', 1070),
 ('Akatsi', 'Harare', 800),
 ('Akatsi', 'Kintampo', 780),
 ('Amanzi', 'Rural', 3100)]

In [133]:
%%sql
#counted number of records per location type

SELECT
    location_type,
    COUNT(location_type) AS records_per_location_type,
FROM
    location
GROUP BY
    location_type
ORDER BY 
    (COUNT(location_type)) DESC;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
2 rows affected.


[('Rural', 23740), ('Urban', 15910)]

In [175]:
%%sql
#calculated the percentage share of water sources in rural vs urban location types
# A major insight derived that most water sources are situated in the rural communities across Maji Ndogo.

SELECT
    ROUND((23740/(23740 + 15910) * 100 )) AS rural_pct_share,
    ROUND((15910/(23740 + 15910) * 100 )) AS urban_pct_share
;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
1 rows affected.


[(Decimal('60'), Decimal('40'))]

## 4. Diving into the sources

In [144]:
%%sql
#quick look on water_source table

SELECT *
FROM
    water_source
LIMIT 5;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
5 rows affected.


[('AkHa00000224', 'tap_in_home', 956),
 ('AkHa00001224', 'tap_in_home_broken', 930),
 ('AkHa00002224', 'tap_in_home_broken', 486),
 ('AkHa00003224', 'well', 364),
 ('AkHa00004224', 'tap_in_home_broken', 942)]

In [174]:
%%sql
# calculated total number of people surveyed

SELECT
    SUM(number_of_people_served) AS total_people_surveyed
FROM
    water_source
;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
1 rows affected.


[(Decimal('27628140'),)]

In [176]:
%%sql
# counted number of each water source

SELECT
    type_of_water_source,
    COUNT(type_of_water_source) AS number_of_water_sources
FROM
    water_source
GROUP BY
    type_of_water_source
ORDER BY
    (COUNT(type_of_water_source)) DESC;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
5 rows affected.


[('well', 17383),
 ('tap_in_home', 7265),
 ('tap_in_home_broken', 5856),
 ('shared_tap', 5767),
 ('river', 3379)]

In [177]:
%%sql
# calculated average number of people using various types of water sources
# For tap_in_home and tap_in_home_broken, 6 people share 1 tap. This is because there is an average of 6 people living in a household.
# This means 1 tap_in_home actually represents 644/6 = ± 100 taps

SELECT
    type_of_water_source,
    ROUND(AVG(number_of_people_served)) AS average_number_of_people_served
FROM
    water_source
GROUP BY
    type_of_water_source
ORDER BY
    ROUND(AVG(number_of_people_served)) DESC;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
5 rows affected.


[('shared_tap', Decimal('2071')),
 ('river', Decimal('699')),
 ('tap_in_home_broken', Decimal('649')),
 ('tap_in_home', Decimal('644')),
 ('well', Decimal('279'))]

In [178]:
%%sql
# calculated total number of people using various types of water sources

SELECT
    type_of_water_source,
    SUM(number_of_people_served) AS total_people_served
FROM
    water_source
GROUP BY
    type_of_water_source
ORDER BY
    SUM(number_of_people_served) DESC;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
5 rows affected.


[('shared_tap', Decimal('11945272')),
 ('well', Decimal('4841724')),
 ('tap_in_home', Decimal('4678880')),
 ('tap_in_home_broken', Decimal('3799720')),
 ('river', Decimal('2362544'))]

In [180]:
%%sql
# calculated percentage share of people using various types of water sources

SELECT
    type_of_water_source,
    ROUND((SUM(number_of_people_served)/(11945272 + 4841724 + 4678880 + 3799720 + 2362544)) * 100) AS pct_share_people_served
FROM
    water_source
GROUP BY
    type_of_water_source
ORDER BY
    ROUND((SUM(number_of_people_served)/(11945272 + 4841724 + 4678880 + 3799720 + 2362544)) * 100) DESC;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
5 rows affected.


[('shared_tap', Decimal('43')),
 ('well', Decimal('18')),
 ('tap_in_home', Decimal('17')),
 ('tap_in_home_broken', Decimal('14')),
 ('river', Decimal('9'))]

## 5. Start of a solution

In [194]:
%%sql
# ranked water sources according to total number of people served, excluding tap_in_home

SELECT
    type_of_water_source,
    SUM(number_of_people_served) AS total_people_served,
    RANK() OVER (ORDER BY (SUM(number_of_people_served)) DESC) AS usage_rank
FROM
    water_source
WHERE
    type_of_water_source != 'tap_in_home'
GROUP BY
    type_of_water_source;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
4 rows affected.


[('shared_tap', Decimal('11945272'), 1),
 ('well', Decimal('4841724'), 2),
 ('tap_in_home_broken', Decimal('3799720'), 3),
 ('river', Decimal('2362544'), 4)]

In [22]:
%%sql
# used rank() to assign a priority rank for each water sources according to number of people served

SELECT
    source_id,
    type_of_water_source,
    number_of_people_served,
    RANK() OVER (PARTITION BY type_of_water_source ORDER BY number_of_people_served DESC) AS priority_rank
FROM 
    water_source
WHERE
    type_of_water_source != 'tap_in_home'
LIMIT 10;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
10 rows affected.


[('SoRu34798224', 'river', 998, 1),
 ('SoRu35837224', 'river', 998, 1),
 ('SoRu36238224', 'river', 998, 1),
 ('SoRu36791224', 'river', 998, 1),
 ('SoRu36880224', 'river', 998, 1),
 ('SoRu38142224', 'river', 998, 1),
 ('SoRu37756224', 'river', 998, 1),
 ('SoMa33775224', 'river', 998, 1),
 ('KiRu30353224', 'river', 998, 1),
 ('SoIl32972224', 'river', 998, 1)]

In [20]:
%%sql
# used dense_rank() to assign a priority rank for each water sources according to number of people served

SELECT
    source_id,
    type_of_water_source,
    number_of_people_served,
    DENSE_RANK() OVER (PARTITION BY type_of_water_source ORDER BY number_of_people_served DESC) AS priority_rank
FROM 
    water_source
WHERE
    type_of_water_source != 'tap_in_home'
LIMIT 10;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
10 rows affected.


[('SoRu34798224', 'river', 998, 1),
 ('SoRu35837224', 'river', 998, 1),
 ('SoRu36238224', 'river', 998, 1),
 ('SoRu36791224', 'river', 998, 1),
 ('SoRu36880224', 'river', 998, 1),
 ('SoRu38142224', 'river', 998, 1),
 ('SoRu37756224', 'river', 998, 1),
 ('SoMa33775224', 'river', 998, 1),
 ('KiRu30353224', 'river', 998, 1),
 ('SoIl32972224', 'river', 998, 1)]

In [23]:
%%sql
# used row_number() to assign a priority rank for each water sources according to number of people served
# Best option to rank since we have ordered the sources from highest priority to lowest priority. 
# We can begin fixing the water sources from the first one and cancel each after completion.
SELECT
    source_id,
    type_of_water_source,
    number_of_people_served,
    ROW_NUMBER() OVER (PARTITION BY type_of_water_source ORDER BY number_of_people_served DESC) AS priority_rank
FROM 
    water_source
WHERE
    type_of_water_source != 'tap_in_home'
limit 10;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
10 rows affected.


[('SoRu34798224', 'river', 998, 1),
 ('SoRu35837224', 'river', 998, 2),
 ('SoRu36238224', 'river', 998, 3),
 ('SoRu36791224', 'river', 998, 4),
 ('SoRu36880224', 'river', 998, 5),
 ('SoRu38142224', 'river', 998, 6),
 ('SoRu37756224', 'river', 998, 7),
 ('SoMa33775224', 'river', 998, 8),
 ('KiRu30353224', 'river', 998, 9),
 ('SoIl32972224', 'river', 998, 10)]

## 6. Analysing queues

In [296]:
%%sql
# calculated the time the visits survey took

SELECT
    MIN(time_of_record) AS min_time_of_record,
    MAX(time_of_record)AS max_time_of_record,
    TIMESTAMPDIFF(day,(MIN(time_of_record)),(MAX(time_of_record))) survey_time_in_days
FROM
    visits;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
1 rows affected.


[(datetime.datetime(2021, 1, 1, 9, 10), datetime.datetime(2023, 7, 14, 13, 53), 924)]

In [310]:
%%sql
# Found out how long, on average, people have to queue for water in Maji Ndogo

SELECT
    AVG(NULLIF(time_in_queue,0)) AS avg_queue_time_in_mins
FROM
    visits;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
1 rows affected.


[(Decimal('123.2574'),)]

In [324]:
%%sql
# uncovered the queue time in different days of the week

SELECT
    DAYNAME(time_of_record) AS day_of_the_week,
    ROUND(AVG(NULLIF(time_in_queue,0))) AS avg_queue_time_in_mins
FROM
    visits
GROUP BY
 DAYNAME(time_of_record)
ORDER BY
    ROUND(AVG(NULLIF(time_in_queue,0))) DESC;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
7 rows affected.


[('Saturday', Decimal('246')),
 ('Monday', Decimal('137')),
 ('Friday', Decimal('120')),
 ('Tuesday', Decimal('108')),
 ('Thursday', Decimal('105')),
 ('Wednesday', Decimal('97')),
 ('Sunday', Decimal('82'))]

In [336]:
%%sql
# uncovered the queue time on an hourly basis during the day
# result is hour of the day, the exact time is calculated in the query that follows
SELECT
    HOUR(time_of_record) AS hour_of_day,
    ROUND(AVG(NULLIF(time_in_queue,0))) AS avg_queue_time_in_mins
FROM
    visits
GROUP BY
    HOUR(time_of_record)
ORDER BY
    ROUND(AVG(NULLIF(time_in_queue,0))) DESC;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
14 rows affected.


[(19, Decimal('168')),
 (7, Decimal('149')),
 (8, Decimal('149')),
 (17, Decimal('149')),
 (6, Decimal('149')),
 (18, Decimal('147')),
 (9, Decimal('118')),
 (13, Decimal('115')),
 (10, Decimal('114')),
 (14, Decimal('114')),
 (15, Decimal('114')),
 (16, Decimal('114')),
 (12, Decimal('112')),
 (11, Decimal('111'))]

In [339]:
%%sql
# further drilled into the exact queue time during the day. Readability is enhanced.

SELECT
    TIME_FORMAT(TIME(time_of_record),'%H:00') AS hour_of_day,
    ROUND(AVG(NULLIF(time_in_queue,0))) AS avg_queue_time_in_mins
FROM
    visits
GROUP BY
    TIME_FORMAT(TIME(time_of_record),'%H:00')
ORDER BY
    ROUND(AVG(NULLIF(time_in_queue,0))) DESC;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
14 rows affected.


[('19:00', Decimal('168')),
 ('07:00', Decimal('149')),
 ('08:00', Decimal('149')),
 ('17:00', Decimal('149')),
 ('06:00', Decimal('149')),
 ('18:00', Decimal('147')),
 ('09:00', Decimal('118')),
 ('13:00', Decimal('115')),
 ('10:00', Decimal('114')),
 ('14:00', Decimal('114')),
 ('15:00', Decimal('114')),
 ('16:00', Decimal('114')),
 ('12:00', Decimal('112')),
 ('11:00', Decimal('111'))]

In [354]:
%%sql
# created a pivot table that calculated the average queue times at different times of days of the week

SELECT
    TIME_FORMAT(TIME(time_of_record), '%H:00') AS hour_of_day,

    # Sunday
    ROUND(AVG(
        CASE
            WHEN DAYNAME(time_of_record) = 'Sunday' THEN time_in_queue
            ELSE NULL
        END 
            ),0) AS Sunday,

    # Monday
    ROUND(AVG(
        CASE
            WHEN DAYNAME(time_of_record) = 'Monday' THEN time_in_queue
            ELSE NULL
        END 
            ),0) AS Monday,

    # Tuesday
    ROUND(AVG(
        CASE
            WHEN DAYNAME(time_of_record) = 'Tuesday' THEN time_in_queue
            ELSE NULL
        END 
            ),0) AS Tuesday,

    # Wednesday
    ROUND(AVG(
        CASE
            WHEN DAYNAME(time_of_record) = 'Wednesday' THEN time_in_queue
            ELSE NULL
        END 
            ),0) AS Wednesday,

    # Thursday
    ROUND(AVG(
        CASE
            WHEN DAYNAME(time_of_record) = 'Thursday' THEN time_in_queue
            ELSE NULL
        END 
            ),0) AS Thursday,

    # Friday
    ROUND(AVG(
        CASE
            WHEN DAYNAME(time_of_record) = 'Friday' THEN time_in_queue
            ELSE NULL
        END 
            ),0) AS Friday,

    # Saturday
    ROUND(AVG(
        CASE
            WHEN DAYNAME(time_of_record) = 'Saturday' THEN time_in_queue
            ELSE NULL
        END 
            ),0) AS Saturday
    
FROM
    visits
WHERE
    time_in_queue != 0
GROUP BY
    hour_of_day
ORDER BY
    hour_of_day;

 * mysql+pymysql://root:***@localhost:3306/md_water_services
14 rows affected.


[('06:00', Decimal('79'), Decimal('190'), Decimal('134'), Decimal('112'), Decimal('134'), Decimal('153'), Decimal('247')),
 ('07:00', Decimal('82'), Decimal('186'), Decimal('128'), Decimal('111'), Decimal('139'), Decimal('156'), Decimal('247')),
 ('08:00', Decimal('86'), Decimal('183'), Decimal('130'), Decimal('119'), Decimal('129'), Decimal('153'), Decimal('247')),
 ('09:00', Decimal('84'), Decimal('127'), Decimal('105'), Decimal('94'), Decimal('99'), Decimal('107'), Decimal('252')),
 ('10:00', Decimal('83'), Decimal('119'), Decimal('99'), Decimal('89'), Decimal('95'), Decimal('112'), Decimal('259')),
 ('11:00', Decimal('78'), Decimal('115'), Decimal('102'), Decimal('86'), Decimal('99'), Decimal('104'), Decimal('236')),
 ('12:00', Decimal('78'), Decimal('115'), Decimal('97'), Decimal('88'), Decimal('96'), Decimal('109'), Decimal('239')),
 ('13:00', Decimal('81'), Decimal('122'), Decimal('97'), Decimal('98'), Decimal('101'), Decimal('115'), Decimal('242')),
 ('14:00', Decimal('83'), Decimal('127'), Decimal('104'), Decimal('92'), Decimal('96'), Decimal('110'), Decimal('244')),
 ('15:00', Decimal('83'), Decimal('126'), Decimal('104'), Decimal('88'), Decimal('92'), Decimal('110'), Decimal('248')),
 ('16:00', Decimal('83'), Decimal('127'), Decimal('99'), Decimal('90'), Decimal('99'), Decimal('109'), Decimal('251')),
 ('17:00', Decimal('79'), Decimal('181'), Decimal('135'), Decimal('121'), Decimal('129'), Decimal('151'), Decimal('251')),
 ('18:00', Decimal('80'), Decimal('174'), Decimal('122'), Decimal('113'), Decimal('132'), Decimal('158'), Decimal('240')),
 ('19:00', Decimal('127'), Decimal('159'), Decimal('145'), Decimal('176'), Decimal('137'), Decimal('103'), Decimal('282'))]

# Conclusion
We uncover these patterns:
1. Queues are very long on a Monday morning and Monday evening as people rush to get water.
2. Wednesday has the lowest queue times, but long queues on Wednesday evening.
3. People have to queue pretty much twice as long on Saturdays compared to the weekdays. It looks like people spend their Saturdays queueing
for water, perhaps for the week's supply?
4. The shortest queues are on Sundays, and this is a cultural thing. The people of Maji Ndogo prioritise family and religion, so Sundays are spent
with family and friends.


# Water Accessibility and infrastructure summary report
This survey aimed to identify the water sources people use and determine both the total and average number of users for each source.
Additionally, it examined the duration citizens typically spend in queues to access water.
Here is a short summary report we can send off to Pres. Naledi:

# Insights
1. Most water sources are rural.
2. 43% of our people are using shared taps. 2000 people often share one tap.
3. 31% of our population has water infrastructure in their homes, but within that group, 45% face non-functional systems due to issues with pipes,
pumps, and reservoirs.
4. 18% of our people are using wells of which, but within that, only 28% are clean..
5. Our citizens often face long wait times for water, averaging more than 120 minutes.
6. In terms of queues:
- Queues are very long on Saturdays.
- Queues are longer in the mornings and evenings.
- Wednesdays and Sundays have the shortest queues.

